File: app/src/main/java/ru/mamykin/foreignbooksreader/views/BooksStoreView.java
package ru.mamykin.foreignbooksreader.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import ru.mamykin.foreignbooksreader.models.StoreBook;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface BooksStoreView extends MvpView {
    void showBooks(List<StoreBook> booksList);

    void showMessage(String message);

    void showLoading(boolean show);
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/presentation/store/BooksStoreView.kt
package ru.mamykin.foreignbooksreader.presentation.store

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

import ru.mamykin.foreignbooksreader.models.StoreBook

@StateStrategyType(AddToEndSingleStrategy::class)
interface BooksStoreView : MvpView {
    fun showBooks(booksList: List<StoreBook>)

    fun showMessage(message: String)

    fun showLoading(show: Boolean)
}